packages/orchestrator-ui-components/src/components/WFOWorkflowSteps/WFOWorkflowStepList/WFOWorkflowStepList.tsx

```diff
@@ -58,25 +75,27 @@ export const WFOWorkflowStepList = React.forwardRef(
                     showHiddenKeys={showHiddenKeys}
                     showRaw={showRaw}
                     allDetailToggleText={
-                        Array.from(stepDetailStates).every((s) => s[1])
-                            ? t('collapseAll')
-                            : t('expandAll')
+                        allStepsAreExpanded ? t('collapseAll') : t('expandAll')
```

Comment:
I'm not sure if this part of the ticket but we have something about this in the loose ends ticket. I think there should be 2 toggles. One for exand all and another for collapsing. This will show expand all even if there is only one item collapsed although you might still want to collapse all;

Reply:
yes, and for now i kept the same logic as it was